include/nil/blueprint/parser.hpp

```diff
@@ -542,10 +577,105 @@ namespace nil {
                             else
                                 base_frame.vectors[current_arg][j] = input_var;
                         }
-                    } else {
-                        assert(llvm::isa<llvm::GaloisFieldType>(arg_type));
-                        assignmnt.public_input(0, public_input_counter) = public_input[public_input_counter];
-                        variables[current_arg] = var(0, public_input_counter++, false, var::column_type::public_input);
+
+                    } else if (llvm::isa<llvm::EllipticCurveType>(arg_type)) {
+                        size_t arg_len = 0;
+
+                        switch (llvm::cast<llvm::EllipticCurveType>(arg_type)->getCurveKind()) {
```

Comment:
Do we need switch here when all the curve types will be supported? If no, please replace it with
```
auto curve_kind = llvm::cast<llvm::EllipticCurveType>(arg_type)->getCurveKind();
assert(curve_kind != ELLIPTIC_CURVE_CURVE25519 && curve_kind != ELLIPTIC_CURVE_BLS12381 && "Unsupported curve kind");
arg_len = 2;